Modules/Core/Common/src/itkObjectFactoryBase.cxx

```diff
@@ -237,14 +238,19 @@ ObjectFactoryBase::Initialize()
 {
   itkInitGlobalsMacro(PimplGlobals);
 
-  // Atomically set m_Initialized to true. If it was false before, enter the if.
-  if (!m_PimplGlobals->m_Initialized.exchange(true))
+  if (!m_PimplGlobals->m_Initialized)
   {
-    ObjectFactoryBase::InitializeFactoryList();
-    ObjectFactoryBase::RegisterInternal();
+    // only one thread initializing at a time, wait until initialization is completed.
+    std::lock_guard lockGuard(m_PimplGlobals->m_Lock);
+    if (!m_PimplGlobals->m_Initialized)
+    {
+      ObjectFactoryBase::InitializeFactoryList();
```

Comment:
Please consider removing the `ObjectFactoryBase::InitializeFactoryList()` call, as in:
- #4225